### PR TITLE
Introduce employee menu same page links

### PR DESCRIPTION
**Pull request by sowbiba**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Employee menu links always have target="_blank" but what if we want to have internal page link? This PR adds a property to allow opening the link in the same page
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | An easy way is to use MBO which adds links to the employee menu. The hard but independant is to have an example module adding links to the employee menu
| Fixed issue or discussion?     | Fixes [https://github.com/PrestaShop/PrestaShop/discussions/33742](https://github.com/PrestaShop/PrestaShop/discussions/33742)
| Related PRs       | -
| Sponsor company   | MBO

### CODE REVIEW

**Comment by sowbiba** [comment]
Check https://github.com/PrestaShop/PrestaShop/discussions/33742

**Review by kpodemski — DISMISSED** [review]

**Comment by sowbiba** [comment]
Maybe too much with the icons ?

Capture d’écran du 2023-08-29 11-37-12

**Comment by TristanLDD** [comment]
Hi @sowbiba! 
I understand this might look a bit repetitive, but differentiating target="blank" links with a specific icon is a good practice I'd like to keep applying everywhere in the BO for consistency purposes.     

Can you check icon's vertical alignment with the text tough? it looks a bit too high 
Thank you !

**Comment by sowbiba** [comment]
> Hi @sowbiba! 
> I understand this might look a bit repetitive, but differentiating target="blank" links with a specific icon is a good practice I'd like to keep applying everywhere in the BO for consistency purposes.     
> 
> Can you check icon's vertical alignment with the text tough? it looks a bit too high 
> Thank you ! 

Hi @TristanLDD 
I limited the scope of the icon to the employee menu links, not for all the BO, because I cannot measure the impact

Ok for the vertical alignment

Capture d’écran du 2023-08-31 13-58-14

**Review by M0rgan01 — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]
LGTM, thanks @sowbiba!

**Review by djoelleuch — COMMENTED** [review]
Hello @sowbiba ,

I have checked your PR and I noticed that we do not have the same behavior before and after clicking

https://github.com/PrestaShop/PrestaShop/assets/115152460/a962d46b-205c-41ac-855d-751c932f3cfc


**Before Clicking**

image

**After Clicking**

image

Could you checked please?

Thank you.

**Comment by sowbiba** [comment]
Fixed @djoelleuch

File: tests/Integration/Controllers/AdminSearchControllerCoreTest.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function dataProviderSearch(): array
                             'title' => 'Search docs.prestashop-project.org',
                             'button_label' => 'Go to the documentation',
                             'link' => 'https://docs.prestashop-project.org/welcome/?q=',
-                            'same_page' => false,
+                            'is_external_link' => false,
```
Comment (jolelievre) [inline tests/Integration/Controllers/AdminSearchControllerCoreTest.php]:
```suggestion
                            'is_external_link' => true,
```

File: tests/Integration/Controllers/AdminSearchControllerCoreTest.php [hunk 2]
```diff
@@ -95,7 +95,7 @@ public function dataProviderSearch(): array
                             'title' => 'Search docs.prestashop-project.org',
                             'button_label' => 'Go to the documentation',
                             'link' => 'https://docs.prestashop-project.org/welcome/?q=orders',
-                            'same_page' => false,
+                            'is_external_link' => false,
```
Comment (jolelievre) [inline tests/Integration/Controllers/AdminSearchControllerCoreTest.php]:
```suggestion
                            'is_external_link' => true,
```

File: src/Core/Search/SearchPanel.php [hunk 3]
```diff
@@ -52,20 +52,20 @@ class SearchPanel implements SearchPanelInterface
...
-    private $isOnSamePage;
+    private $isExternalLink;
 
     public function __construct(
         string $title,
         string $buttonLabel,
         string $link,
         array $queryParams,
-        ?bool $isOnSamePage = false
+        ?bool $isExternalLink = false
```
Comment (jolelievre) [inline src/Core/Search/SearchPanel.php]:
```suggestion
        ?bool $isExternalLink = true
```

**Review by kpodemski — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]
Thanks @sowbiba

**Review by aniszr — COMMENTED** [review]
Hello @sowbiba 

I tested your PR and I mentioned that the same issue still present as mentioned before!

**After first click :**

userprofile1

After you enter to another page the page display became :

userprofile2

Thanks!

**Comment by sowbiba** [comment]
Sorry but please compare to 8.1.1 (without the PR)
The differences you both mentioned were already there. The only thing to observe here is the presence of the icons and the fact that links are opened in the same page

**Review by boherm — DISMISSED** [review]

**Review by aniszr — COMMENTED** [review]
Hello @sowbiba 

Could you please check these behaviors when clicking on employer links :

- Training, Find an expert ,Help Center links opens new tabs in browser and redirect to external links :x:
- Only Prestashop Marketplace is displayed in same page :heavy_check_mark: 

See attached screenshot:

image

For visual aspect, could you please check why the navigation icon is no more displayed when you open any page from shop BO

See attached screenshot :

image

Thank you!

**Comment by sowbiba** [comment]
@aniszr You are right. There was a missing piece of CSS

Fixed

**Comment by sowbiba** [comment]
@aniszr You can use 
[ps_mbo.zip](https://github.com/PrestaShop/PrestaShop/files/12618285/ps_mbo.zip)
for testing

**Review by jolelievre — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Review by aniszr — COMMENTED** [review]
Hello @sowbiba 

QA :heavy_check_mark: 

Waiting for Auto Test!

Thank you!

**Review by aniszr — APPROVED** [review]
Hello @sowbiba 
I tested the improvements you added to fix the mentioned issues:

- The icon display is fixed :heavy_check_mark:
See attached screenshot:

image

- Links with redirect icons are redirected with an external link to a new tab :heavy_check_mark:
- Links with no redirection icon are opened in local page :heavy_check_mark:

Auto Test : https://github.com/aniszr/ga.tests.ui.pr/actions/runs/6199934095 :heavy_check_mark:  (1 Test failed mentioned in Nightly)

Thank you!

**Pull request merged**